full-service/src/service/wallet_impl.rs

```diff
@@ -98,16 +106,129 @@ impl<
             fog_pubkey_resolver,
             _sync_thread: sync_thread,
             submit_node_offset: Arc::new(AtomicUsize::new(rng.next_u64() as usize)),
+            offline,
             logger,
         }
     }
 
+    // Helper method to expand password to password hash using argon2.
+    fn get_password_hash(
+        &self,
+        password: Option<String>,
```

Comment:
Why bother with supporting both? This is a pattern I have never seen anywhere else. I think should just accept `password` and always hash it.

Reply:
We want the mobile and desktop wallet to allow configurable Argon2 parameters so the user can control time of encryption (2s, 5s, 10s, etc.) This is important based on the processing speed of their devices (specifically some phones).

That logic doesn't seem right within Full-Service.

The docs should say that 99% of users will want to give it the `password: <string>` option. And you should only use the `passwordHash: <hex encoded string>` if you know why you're using it.